pymc4/inference/utils.py

```diff
@@ -31,23 +31,49 @@ def initialize_sampling_state(
     return state, deterministic_names + transformed_names
 
 
-def trace_to_arviz(pm4_trace, pm4_sample_stats):
+def trace_to_arviz(
+    trace=None,
+    sample_stats=None,
+    observed_data=None,
+    prior_predictive=None,
+    posterior_predictive=None,
+    inplace=True,
+):
     """
     Tensorflow to Arviz trace convertor.
 
-    Convert a PyMC4 trace as returned by sample() to an ArviZ trace object
-    that can be passed to e.g. arviz.plot_trace().
+    Creates an ArviZ's InferenceData object with inference, prediction and/or sampling data
+    generated by PyMC4
 
     Parameters
     ----------
-    pm4_trace : dict
-    pm4_sample_stats : dict
+    trace : dict or InferenceData
+    sample_stats : dict
+    observed_data : dict
+    prior_predictive : dict
+    posterior_predictive : dict
+    inplace : bool
 
     Returns
     -------
-    arviz.data.inference_data.InferenceData
+    ArviZ's InferenceData object
     """
+    if trace is not None and isinstance(trace, dict):
+        trace = {k: np.swapaxes(v.numpy(), 1, 0) for k, v in trace.items() if "/" in k}
```

Comment:
The `if "/" in k` prevents adding to the posterior group the observed data. observed data  it's added to its own group. I have not found an example where this becomes problematic. Could you provide a short example of such scenario?

Reply:
The `if "/" in k` doesn't prevent adding the observed data to the posterior group. The observed data just isn't returned by `sample` because it only collects the values of transformed, untransformed and deterministics variables. In the following model:

```python
@pm.model:
def model()
    mu = yield pm.Normal(loc=0, scale=1, name="mu")
    observed = yield pm.Normal(loc=mu, scale=1, name="observed", observed=np.zeros(10))
    return observed
```

Evaluating the model will give you the following variables: "model/mu", "model/observed", "model". The last "model" variable is stored as a deterministic, and that means that it will be returned in the `sample`ed trace. The `if "/" in k` means that the deterministic variable named "model" will be dropped. It doesn't look like a big deal here, but it isn't transparent to the user. 

Furthermore, if the users pass a coroutine directly to the prior predictive sampler, no "model/" will be prepended, and all the root level variables will be dropped.

Reply:
"model" is dropped here, because observed data has it's own group and it does not belong to the posterior group. see https://github.com/pymc-devs/pymc4/pull/199/files#diff-2eb557123ba49614ee42a55d2dae7fdfR160

I have removed the `if "/" in k` from the prior_predictive.